example-app/src/main/java/com/adyen/checkout/example/ui/card/CardViewModel.kt

```diff
@@ -89,8 +89,7 @@ internal class CardViewModel @Inject constructor(
                 amount = keyValueStorage.getAmount(),
                 countryCode = keyValueStorage.getCountry(),
                 merchantAccount = keyValueStorage.getMerchantAccount(),
-                // Should be set correctly for additional actions
-                redirectUrl = "",
+                redirectUrl = "adyencheckout://com.adyen.checkout.example/card",
```

Comment:
We usually get this URL by calling `RedirectComponent.getReturnUrl(applicationContext)`. Maybe you can try "${RedirectComponent.getReturnUrl(applicationContext)}/card"? You need to make your view model an `AndroidViewModel` but that's alright I guess?

Reply:
Tbh I don't really like `AndroidViewModel`, because it makes unit testing difficult. Not that we have tests at the moment, but you get the point 😅 There are other ways to use `RedirectComponent.getReturnUrl` without using `AndroidViewModel`, but I didn't want to spent too much time on it

Reply:
Fair enough 👍